File: order-service/src/main/java/com/intexsoft/webshop/orderservice/controller/OrderController.java
package com.intexsoft.webshop.orderservice.controller;

import com.intexsoft.webshop.orderservice.dto.order.OrderCreateDto;
import com.intexsoft.webshop.orderservice.dto.order.OrderDto;
import com.intexsoft.webshop.orderservice.service.OrderService;
import com.intexsoft.webshop.orderservice.util.JsonUtils;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/order")
@RequiredArgsConstructor
@Slf4j
@Validated
public class OrderController {
    private final OrderService orderService;

    @PostMapping
    public ResponseEntity<OrderDto> createOrder(@RequestBody @Valid OrderCreateDto orderCreateDto) {
        log.info("IN: request to create a new order received. Request body = {}",
                JsonUtils.getAsString(orderCreateDto));
        OrderDto createdOrderDto = orderService.saveOrder(orderCreateDto);
        log.info("OUT: new order created successfully. Response body = {}", JsonUtils.getAsString(createdOrderDto));
        return ResponseEntity.ok(createdOrderDto);
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/controller/OrderController.kt
package com.intexsoft.webshop.orderservicekt.controller

import com.intexsoft.webshop.orderservicekt.dto.order.OrderCreateDto
import com.intexsoft.webshop.orderservicekt.dto.order.OrderDto
import com.intexsoft.webshop.orderservicekt.log
import com.intexsoft.webshop.orderservicekt.service.OrderService
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/order")
@Validated
class OrderController(
    private val orderService: OrderService
) {
    @PostMapping
    fun createOrder(@RequestBody orderCreateDto: @Valid OrderCreateDto): ResponseEntity<OrderDto> {
        log.info("IN: request to create a new order received. Request body = $orderCreateDto")
        val createdOrderDto: OrderDto = orderService.saveOrder(orderCreateDto)
        log.info("OUT: new order created successfully. Response body = $createdOrderDto")
        return ResponseEntity.ok(createdOrderDto)
    }
}